### PR TITLE
Remove vendored Boost and add Nlohmann JSON

### PR DESCRIPTION
**Pull request by oleorhagen**
* Removes boost vendor dir
* Add nlohmann/json as a submodule (header only json lib)
* Integrate our common directory with the nlohmann/json library

Note: I have not removed the m4 boost files. We will need them sooner or later. If you think this is dirty, speak up.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -2,6 +2,6 @@
 	path = vendor/googletest
 	url = https://github.com/google/googletest.git
 	branch = v1.12.x
-[submodule "vendor/boost"]
-	path = vendor/boost
-	url = https://github.com/boostorg/boost.git
+[submodule "vendor/json"]
```
Comment (oleorhagen) [inline .gitmodules:5]: I named it vendor/json... Not sure if this is completely kosher or not.

File: configure.ac [hunk 2]
```diff
@@ -18,38 +18,16 @@ dnl (Make these variables available for use in Makefile.am)
...
+dnl TODO: Check which is the actual minimum we support!
+AX_BOOST_BASE([1.74.0],
+    [AC_SUBST([boost_CPPFLAGS], [${BOOST_CPPFLAGS}])
+        AC_MSG_NOTICE([boost_CPPFLAGS : ${boost_CPPFLAGS}])
+        AC_MSG_NOTICE([boost_LDFLAGS : ${boost_LDFLAGS}])],
+    [AC_MSG_ERROR([Boost 1.74.0 or later is required but was not found.])])
+
+AX_BOOST_JSON
+AC_SUBST([boost_json_LIBS], [${BOOST_JSON_LIB}])
+AC_MSG_NOTICE([boost_json_LIBS : ${boost_json_LIBS}])
```
Comment (vpodzime) [inline configure.ac:30]: Revert (`reset -p`?) artifact here.

Comment (oleorhagen) [reply configure.ac]: Hmm, I'm not following. What do you mean?

Comment (vpodzime) [reply configure.ac]: Oh, I thought you were removing it completely. But you are, **properly**, first just making it not vendored. :+1:

File: common/json/impl/nlohmann/nlohmann_json.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+#include <nlohmann/json.hpp>
+using njson = nlohmann::json;
+
+#include <common/json/impl/nlohmann/nlohmann_json.hpp>
+
+namespace json {
+
+void NlohmannJson::hello_world() {
+	std::ifstream f("example.json");
+	njson data = njson::parse(f);
```
Comment (vpodzime) [inline common/json/impl/nlohmann/nlohmann_json.cpp:26]: Could we use `istream` instead? Otherwise we need the file with proper content.

**Review by vpodzime — COMMENTED** [review]
Looks good to me otherwise.

File: configure.ac (hunk 2 above)
Comment (oleorhagen) [reply configure.ac]: Yes, I wanted to have every commit be functional still :smiley_cat:

File: common/json/impl/nlohmann/nlohmann_json.cpp (hunk 3 above)
Comment (oleorhagen) [reply common/json/impl/nlohmann/nlohmann_json.cpp:26]: Done!

**Review by vpodzime — APPROVED** [review]
ACK

**Pull request merged**